content/sctk-simple-image-viewer.markdown

```diff
@@ -0,0 +1,657 @@
+            // order of Close > Configure > Refresh.
+            // Indeed, if we received a Close, there is not point drawing anything anymore as
+            // we will exit. A new Configure overrides a previous one, and if we received
+            // a Configure we will refresh the frame anyway.
+            let mut next_action = waction.lock().unwrap();
+            // Check if we need to replace the old event by the new one
+            let replace = match (&evt, &*next_action) {
+                // replace if there is no old event
+                (_, &None)
+                // or the old event is refresh
+                | (_, &Some(WEvent::Refresh))
+                // or we had a configure and received a new one
+                | (&WEvent::Configure { .. }, &Some(WEvent::Configure { .. }))
+                // or the new event is close
+                | (&WEvent::Close, _) => true,
+                // keep the old event otherwise
+                _ => false,
+            };
+            if replace {
+                *next_action = Some(evt);
+            }
+        },
+        // creating the window may fail if the code drawing the frame
+        // fails to initialize itself. For BasicFrame this should not happen
+        // unless the system is utterly broken, though.
+    ).expect("Failed to create a window !");
+
+    // The window needs access to pointer inputs to handle the user interaction
+    // with it: resizing/moving the window, or clicking its button.
+    // user interaction in Wayland is done via the wl_seat object, which represents
+    // a set of pointer/keyboard/touchscreen (some may be absent) that logically
+    // represent a single user. Most systems have only one seat, multi-seat configurations
+    // are quite exotic.
+    // Thus, we just automatically bind the first seat we find.
+    let seat = env.manager
+        .instantiate_auto::<wl_seat::WlSeat>()
+        .unwrap()
+        .implement(move |_, _| {});
+    // And advertize it to the Window so it knows of it and can process the
+    // required pointer events.
+    window.new_seat(&seat);
+
+    // ..
+}
+```
+
+At this point, we have created most of the Wayland objects we'll need for this app.
+It's time to jump to the next step:
+
+### Drawing the contents
+
+In here we'll intialize the few more resources we need to do our drawing, and define
+a function that redraws our window for a given size, resizing the loaded image if
+necessary.
+
+Defining the contents of a surfaces is done in Wayland by first drawing the contents
+ourselves, creating a `wl_buffer` referencing this content, and then attaching the
+buffer to the surface. The two mains ways of sharing buffers is via OpenGL or using
+a shared memory in case of CPU drawing. We'll do the later.
+
+A shared memory pool in Wayland context is just a file (more precisely a tempfile, that
+exists only in memory, not on disk) in which the client writes some content and shares the
```

Comment:
into which

Reply:
From "and shares" on this sentence reads weirdly. Can't recommend a small fix off the top of my head though.